runelite-api/src/main/java/net/runelite/api/Perspective.java

```diff
@@ -771,4 +772,119 @@ public static Point getCanvasTextMiniMapLocation(
 		return new Point(xOffset, yOffset);
 	}
 
+	/**
+	 * Calculates a polygon line from 2 LocalPoints
+	 *
+	 * @param client the game client
+	 * @param startLocation start location of the polygon
+	 * @param endLocation end location of the polygon
+	 * @return a {@link Polygon}
+	 */
+	private static Polygon linePoly(@Nonnull Client client, @Nonnull LocalPoint startLocation, @Nonnull LocalPoint endLocation)
+	{
+		LocalPoint startPoint = new LocalPoint(
+				startLocation.getX() - (LOCAL_TILE_SIZE / 2),
+				startLocation.getY() + (LOCAL_TILE_SIZE / 2));
+		LocalPoint endPoint = new LocalPoint(
+				endLocation.getX() - (LOCAL_TILE_SIZE / 2),
+				endLocation.getY() + (LOCAL_TILE_SIZE / 2));
+		int plane = client.getPlane();
+		Point p1 = Perspective.localToCanvas(client, startPoint, plane);
+		Point p2 = Perspective.localToCanvas(client, endPoint, plane);
+		if (p1 != null && p2 != null)
+		{
+			Polygon polygon = new Polygon();
+			polygon.addPoint(p1.getX(), p1.getY());
+			polygon.addPoint(p2.getX(), p2.getY());
+			return polygon;
+		}
+		return null;
+	}
+
+	/**
+	 * Calculates a list of polygon lines with 2 Worldpoints
+	 * Start location is the South-West WorldPoint, end location is the North-East WorldPoint
+	 * @param client the game client
+	 * @param startLocation start location of the polygon
+	 * @param endLocation end location of the polygon
+	 * @return a {@link List} of {@link Polygon} 
+	 */
+	public static List<Polygon> getLinePolyList(@Nonnull Client client, @Nonnull WorldPoint startLocation, @Nonnull WorldPoint endLocation)
```

Comment:
Doesn't WorldArea already do this? We do mark some >1x1 areas already eg for npc respawns using it.